apps/src/responsive.jsx

```diff
@@ -0,0 +1,119 @@
+/* A very simple responsive layout system.
+ */
+
+import * as utils from './utils';
+import $ from 'jquery';
+
+export default class Responsive {
+
+  constructor(responsiveWindowWidth) {
+    this.responsiveWindowWidth = responsiveWindowWidth;
+  }
+
+  /**
+   * Gets the container width.
+   * Returns either a number (e.g. 1170) or a string (e.g. "97%").
+   */
+  getResponsiveContainerWidth() {
+    const windowWidth = $(window).width();
+
+    if (windowWidth >= 970+50) {
+      return 970;
+    } else {
+      return "97%";
+      // return Math.round(0.97 * windowWidth);
+    }
+  }
+
+  // makeEnum comes from apps/src/utils
+  static ResponsiveSize = utils.makeEnum('lg', 'md', 'sm', 'xs');
+
+  /**
+   * Returns the window width that is the starting point for a width category.
+   *
+   * @param {string} id - "xs", "sm", "md", or "lg"
+   */
+  getResponsiveWindowWidth(category) {
+    return this.responsiveWindowWidth[category];
+  }
+
+  /**
+   * Returns whether provided category is active, given current window width.
+   * e.g. called with "md" when window width >= 820px returns true.
+   *
+   * @param {string} id - "xs", "sm", "md", or "lg"
+   */
+  isResponsiveCategoryActive(category) {
+    return $(window).width() >= this.responsiveWindowWidth[category];
+  }
+
+  /**
+   * Returns whether provided category is inactive, given current window width.
+   * e.g. called with "md" when window width < 820px returns false.
+   *
+   * @param {string} id - "xs", "sm", "md", or "lg"
+   */
+  isResponsiveCategoryInactive(category) {
+    return $(window).width() < this.responsiveWindowWidth[category];
```

Comment:
so if you're in a `md` screen width and you're checking to see if `xs` is inactive it would return false?

Reply:
Correct.

Reply:
So instead of being inactive are we not checking whether its bigger than the current category? A different name or more detailed comment here would be helpful.

Reply:
I always have to think this through too.  It's inspired by Bootstrap's show/hidden [system](https://getbootstrap.com/docs/3.3/css/#responsive-utilities-classes) but it's a bit more flexible here because the caller decides what it wants to do with the information that a category is active or not.